arangodb-net-standard.Test/Serialization/JsonNetApiClientSerializationTest.cs

```diff
@@ -35,36 +35,62 @@ public void Serialize_ShouldSucceed()
             string jsonStringWithoutCamelCase = Encoding.UTF8.GetString(jsonBytesWithoutCamelCase);
 
             // standard property with and without camelCase
-            Assert.Contains("propertyToCamelCase", jsonStringWithCamelCase);
-            Assert.Contains("PropertyToCamelCase", jsonStringWithoutCamelCase);
+            Assert.Contains("propertyToCheckIfCamelCase", jsonStringWithCamelCase);
+            Assert.Contains("PropertyToCheckIfCamelCase", jsonStringWithoutCamelCase);
 
             // Null property should be ignored in both cases
             // (ignore case is important to make sure we don't miss the string 
             // if it is using different casing than we checked for)
             Assert.DoesNotContain(
-                "nullPropertyToIgnore",
+                "nullProperty",
                 jsonStringWithCamelCase,
                 System.StringComparison.OrdinalIgnoreCase);
 
-            Assert.DoesNotContain("nullPropertyToIgnore", 
+            Assert.DoesNotContain("nullProperty",
                 jsonStringWithoutCamelCase,
                 System.StringComparison.OrdinalIgnoreCase);
 
             // We expect enum conversion to string, as well as camelCase
-            Assert.Contains("enumToConvertToString", jsonStringWithCamelCase);
+            Assert.Contains("enumToConvert", jsonStringWithCamelCase);
 
             // We expect enum conversion to string, but not camelCase
-            Assert.Contains("EnumToConvertToString", jsonStringWithoutCamelCase);
+            Assert.Contains("EnumToConvert", jsonStringWithoutCamelCase);
         }
 
         [Fact]
         public void Serialize_ShouldSucceed_WhenUsingDefaultOptions()
         {
             var model = new TestModel()
             {
-                NullPropertyToIgnore = null,
-                PropertyToCamelCase = "myvalue",
-                EnumToConvertToString = TestModel.Number.Two
+                NullProperty = null,
```

Comment:
I think a comment to explain what the rules are might be nice.  Maybe we should split out the different test cases into separate test functions. i.e. there are a few aspects of serializer behaviour under test (from what I gather):

- camelCase conversion of property names (should convert)
- camelCase conversion of dictionary keys (should not convert?)
- camelCase conversion of dictionary values (should convert?)
- camelCase conversion of nested property names (should convert)
- ignoring properties with null values (should ignore)
- ignoring dictionary entries with null values (should not ignore?)
- converting Enum values to strings (should always convert)

Did I get that right?  Did I miss any out?

We could create separate [Fact]-based tests for each behaviour, or perhaps look into using a [Theory] to setup the same test logic but comparing some different models and the expected results.

Reply:
Probably separate tests would be simpler to be honest

Reply:
I have approved the PR, I'll leave it up to you if you think we should split into separate tests as a separate issue.

Reply:
Separate tests would be useful, yes. I'm checking the rules you listed and I have a hard time remembering all the behaviors. So I think my original description (below) still doesn't apply 😆 

> This is to give a clearer view of what to expect when serializing with default options.

From the rules you listed, here are some edits:

> camelCase conversion of dictionary values (should convert?)

I think so, if the value is an object.

> converting Enum values to strings (should always convert)

Should convert based on specified settings, rather than always.

Reply:
Created #341